json/src/main/java/org/apache/james/json/DTOConverter.java

```diff
@@ -61,6 +62,7 @@ public DTOConverter(Set<DTOModule<T, U>> modules) {
         String type = dto.getType();
         return Optional
             .ofNullable(typeToModule.get(type))
+            .map(module -> (DTOModule<T, U>) module)
```

Comment:
because life is sad ):
because I needed typemode to have covariant types:
 ```Map<String, DTOModule<? extends T, ? extends U>> typeToModule```
and then I can't call `convert(dto)` because ? can be incompatible with the exact type of `dto`